### ISSUE TITLE
no-descending-specificity rule detection

### ISSUE THREAD

**Issue by erosman** [stylelint/stylelint#7321]
### What minimal example or steps are needed to reproduce the bug?

```css
.pattern input:nth-of-type(2) {
  grid-column: span 3;
}

.tester input {
  font-size: 1em;
}
```

### What minimal configuration is needed to reproduce the bug?

```json
{
  "rules": {
    "no-descending-specificity": true
  }
}
```

### How did you run Stylelint?

Official [Stylelint extension](https://marketplace.visualstudio.com/items?itemName=stylelint.vscode-stylelint) for Visual Studio Code

_PS. The demo page doesn't show the error when I tried it._

### Which Stylelint-related dependencies are you using?

```json
{
  "stylelint": "15.11.0",
  "stylelint-config-standard": "33.0.0"
}
```

### What did you expect to happen?

No error

### What actually happened?

> Expected selector ".tester input" to come before selector ".pattern input:nth-of-type(2)" (no-descending-specificity) Stylelint ([no-descending-specificity](https://stylelint.io/user-guide/rules/no-descending-specificity))

### Do you have a proposal to fix the bug?

The `input` rules in the above example are completely unrelated.

**Comment by ybiquitous** [comment]
@erosman Thanks for using the template and providing a reproducible example.

This behavior is expected as the [rule document describes](https://stylelint.io/user-guide/rules/no-descending-specificity/#dom-limitations):

> This is a correct error because the `a:hover` on line 2 has a higher specificity than the `a` on line 3.

In the following example, `font-size: 10px` is applied instead of `font-size: 1em`:

```html
<div class="a">
  <div class="b">
    <input />
  </div>
</div>
```

```css
.a input:nth-of-type(1) { /* Specificity (0,2,1) */
  font-size: 10px; /* win ✅ */
}

.b input { /* Specificity (0,1,1) */
  font-size: 1em; /* lose ❌ */
}
```
